Indicators/HilbertTransformIndicator.cs

```diff
@@ -0,0 +1,162 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using QuantConnect.Logging;
+
+namespace QuantConnect.Indicators;
+
+/// <summary>
+/// This indicator computes the Hilbert Transform Indicator by John Ehlers.
+/// By using present and prior price differences, and some feedback, price values are split into their complex number components
+/// of real (inPhase) and imaginary (quadrature) parts.
+/// http://www.technicalanalysis.org.uk/moving-averages/Ehle.pdf
+/// </summary>
+public class HilbertTransformIndicator : Indicator, IIndicatorWarmUpPeriodProvider
+{
+    private const int Quad2Length = 2;
+    private const int InPhase3Length = 3;
+
+    private readonly IndicatorBase<IndicatorDataPoint> _input;
+    private readonly IndicatorBase<IndicatorDataPoint> _prev;
+    private readonly IndicatorBase<IndicatorDataPoint> _v1;
+    private readonly IndicatorBase<IndicatorDataPoint> _v2;
+    private readonly IndicatorBase<IndicatorDataPoint> _v4;
+    private readonly Queue<IndicatorDataPoint> _inPhase3 = new();
+    private readonly Queue<IndicatorDataPoint> _quad2 = new();
```

Comment:
I suggest replacing the `Queue` for `RollingWindow`.
```csharp
private readonly RollingWindow<IndicatorDataPoint> _quadratureWindow = new RollingWindow<IndicatorDataPoint>(2);
```
where `2` comes from `Quad2Length = 2` and we can remove this field. Same for _inPhase3.

Reply:
fixed by bb5b228f9f3c1fe6d2df4b487cdbcd4c08a501d7